File: services/gameService/GameTile.java
package services.gameService;

import java.awt.Image;
import javax.swing.Icon;
import javax.swing.ImageIcon;
import javax.swing.JLabel;

/**
 *
 * @author Jean Carlos Santoya Cabrera jeancarlosodi@gmail.com
 */
public abstract class GameTile extends JLabel{
    private String title;
    
    public GameTile(int x, int y, String path, int width, int height, String title){
        this.title = title;
        
        ImageIcon iIcono = new ImageIcon(Icon.class.getResource(path));
        Icon icono = new ImageIcon(iIcono.getImage().getScaledInstance(width, height, Image.SCALE_DEFAULT));
        
        setSize(icono.getIconWidth(), icono.getIconHeight());
        setLocation(x, y);
        setIcon(icono);
    }
    
    public String getTitle(){
        return title;
    }
    
}

File: game/GameTile.kt
package services.gameService

import java.awt.Image
import javax.swing.Icon
import javax.swing.ImageIcon
import javax.swing.JLabel

/**
 *
 * @author Jean Carlos Santoya Cabrera jeancarlosodi@gmail.com
 */
abstract class GameTile(x: Int, y: Int, path: String?, width: Int, height: Int, val title: String) : JLabel() {

    init {
        val iIcono = ImageIcon(Icon::class.java.getResource(path))
        val icono: Icon = ImageIcon(iIcono.image.getScaledInstance(width, height, Image.SCALE_DEFAULT))
        setSize(icono.iconWidth, icono.iconHeight)
        setLocation(x, y)
        icon = icono
    }
}